lib/host-sp-messages/src/lib.rs

```diff
@@ -570,6 +601,48 @@ impl Identity {
     pub const SERIAL_LEN: usize = 51;
 }
 
+#[derive(
+    Debug, Clone, Copy, PartialEq, Eq, Deserialize, Serialize, SerializedSize,
+)]
+pub struct Barcode(#[serde(with = "BigArray")] [u8; Self::LEN]);
+
+impl Barcode {
+    pub const LEN: usize = oxide_barcode::Mpn1Identity::MAX_LEN;
+}
+
+impl Default for Barcode {
+    fn default() -> Self {
+        Self([0; Self::LEN])
+    }
+}
+
+impl From<oxide_barcode::Mpn1Identity> for Barcode {
+    fn from(
+        oxide_barcode::Mpn1Identity { buf, .. }: oxide_barcode::Mpn1Identity,
```

Comment:
Nit: this kind of unpacking is slightly too clever for my taste, maybe just `id: oxide_barcode::Mpn1Identity` and then `Self(id.buf)`?